scripts/smi/deploy.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+set -e
+
+if ! type "kubectl" > /dev/null 2>&1; then
+  printf "ERROR\tgrep cannot be found\n"
+  exit 1;
+fi
+
+printf "INFO\tStarting SMI deployment......\n"
+if ! kubectl apply -f ./scripts/smi/smi.yaml; then
```

Comment:
I wonder whether we have some guidelines when to use command line tool, and when to use client in the code, if one exists, like for kubectl? The consul adapter is using the kubectl client. Advantages and disadvantages of each approach, is e.g. error handling and error message propagation easier with a client?

Reply:
@mgfeller I had the same concern. My default perspective is to always use the client built for the language (use Kubernetes client go). 

The crux of the consideration in this circumstance is the lack of a kuma client go, I believe, and the lack of public Kubernetes manifests? I think that we should investigate two things:

1. Availability of Kubernetes manifests for Kuma - and consider using kubernetes go client with these manifests
1. If for whatever reason the Kuma project is not publishing their Kubernetes deployment manifests (which would be _really_ odd), we could still use Kubernetes client go after piping the output of this command to a file / object: `kumactl install control-plane | kubectl apply -f -`

Reply:
@lee ok, good to know, this default perspective makes sense. 
Kuma can be installed using Helm too https://kuma.io/docs/0.7.1/installation/helm/.

If we want to Helm functionality (and other stuff) common to all adapters (as discussed elsewhere), where would that code live? Learn Layer 5 repo like the smi.yaml as mentioned above? Or a new repo? Sorry if I have overlooked that in a doc or on slack :thinking:

Reply:
@mgfeller  Totally agree with the above. I am currently pushing changes for client-go and helm client. Also about the helm chart, we are moving towards supporting multiple ways to deploy the mesh. The user decides which way he wants the mesh to be installed.

Reply:
@kumarabd :+1: looking forward to seeing your code using the helm client, it's on the list for the consul adapter, hopefully able to work on that soon.

Reply:
@mgfeller  The helm changes are pushed.

Reply:
@kumarabd :+1:

Reply:
Adding the default perspectives above as guiding principles in the Meshery Adapters design spec.